packages/backend/src/controllers/tasks/tasks.controller.ts

```diff
@@ -0,0 +1,247 @@
+  async createTask(
+    @Body() createTaskDto: CreateTaskDto,
+    @User() user: DecodedIdToken,
+  ) {
+    const userDoc = await this.userStoreService.findOneByFirebaseId(user.uid);
+
+    if (userDoc.house != undefined) {
+      const house = await this.houseStoreService.findOne(userDoc.house);
+
+      if (house != undefined) {
+        if (createTaskDto.pool.length < 1) {
+          throw new HttpException('pool is empty', HttpStatus.BAD_REQUEST);
+        }
+
+        createTaskDto.assigned = this.taskUtil.selectRandomUser(
+          createTaskDto.pool,
+        );
+        createTaskDto.house = house._id;
+
+        await this.taskStoreService.create(createTaskDto);
+
+        // 201 code will be returned by default
+        return;
+      }
+    }
+
+    throw new HttpException('user is not in a house', HttpStatus.BAD_REQUEST);
+  }
+
+  @Get('/tasks')
+  @ApiOperation({ summary: 'get the current tasks from a house' })
+  @ApiOkResponse({
+    description: 'tasks retrieved successfully',
+    type: HouseTasksResponseDto,
+  })
+  @ApiBadRequestResponse({ description: 'user is not in a house' })
+  async getTasksForHouse(
+    @User() user: DecodedIdToken,
+  ): Promise<HouseTasksResponseDto | null> {
+    const userDoc = await this.userStoreService.findOneByFirebaseId(user.uid);
+
+    if (userDoc.house != undefined) {
+      const house = await this.houseStoreService.findOne(userDoc.house);
+
+      if (house != undefined) {
+        const tasks = await this.taskStoreService.findAll();
+
+        const tasksForHouse = tasks.filter((task) =>
+          task.house.equals(house._id),
+        );
+
+        const tasksDue = this.taskUtil.checkRecurrence(tasksForHouse);
+        tasksDue.forEach(
+          async (task) =>
+            await this.taskStoreService.update(task.id, task.updatedTask),
+        );
+
+        const updatedTasks = await this.taskStoreService.findAll();
+        const updatedTasksForHouse = updatedTasks.filter((task) =>
+          task.house.equals(house._id),
+        );
+        return {
+          tasks: updatedTasksForHouse,
+        };
+      }
+    }
+
+    throw new HttpException('user is not in a house', HttpStatus.BAD_REQUEST);
+  }
+
+  @Delete('/tasks/:id')
+  @HttpCode(204)
+  @ApiOperation({ summary: 'delete a task from a house.' })
+  @ApiNoContentResponse({
+    description: 'task successfully deleted.',
+  })
+  @ApiBadRequestResponse({
+    description: 'user is not the owner of the house',
+  })
+  async deleteTaskFromHouse(@Param('id') id, @User() user: DecodedIdToken) {
+    const userDoc = await this.userStoreService.findOneByFirebaseId(user.uid);
+
+    if (userDoc.house != undefined) {
+      const house = await this.houseStoreService.findOne(userDoc.house);
+      const task = await this.taskStoreService.findOne(id);
+
+      if (house.owner.equals(userDoc._id)) {
+        this.taskStoreService.delete(task._id);
+        return 'task deleted successfully';
+      }
```

Comment:
is it sound logic that only the owner of the house can delete the task? is this what we want? we can give tasls owners for similar logic to this, but I think we are electing to keeping things open so that anyone in the house can delete a task, likewise is the logic in my notes controller where anyone can delete a note

Reply:
What are your thoughts @j-chad ? I agree with andreas